### PR TITLE
Use classnames instead of deprecated classSet, directly export module 

### PR DESCRIPTION
**Pull request by dahei**
Hi,

this is my first gh-fork and pull request, so please bear with me, if I did something wrong.

This pull request addresses 2 issues:

https://github.com/deseretdigital-ui/class-name-mixin/issues/3

I changed the exports to the actual object (without the intermediate variable ClassNameMixin). Because of this 'breaking change' I raised the version to 2.0.0

https://github.com/deseretdigital-ui/class-name-mixin/issues/4

I replaced the classSet with the recommended classnames module:

https://github.com/JedWatson/classnames

The tests run fine, I tweaked the gulp tasks a litte bit to get the example running.

Thx,
Daniel

### CODE REVIEW

**Comment by rsahlstrom** [comment]
Thanks so much for this pull request and the issues. Let me look over this and get it merged and a new release tagged. I should have time to do that sometime this weekend.

**Comment by rsahlstrom** [comment]
Running `gulp buildExample` fails for me. Did you forget to commit /example/index.js?

File: src/mixins/ClassNameMixin.jsx [hunk 1]
```diff
@@ -42,7 +43,7 @@ var ClassNameMixin = {
     var mergedClassName = mergeClassSetObjects(componentClassName, className);
     mergedClassName = mergeClassSetObjects(mergedClassName, propClassName);
 
-    return React.addons.classSet(mergedClassName);
+    return classNames(mergedClassName);
```
Comment (rsahlstrom) [inline src/mixins/ClassNameMixin.jsx:46]: Now that we're using classNames, we no longer need react/addons and we can change:

``` js
var React = require('react/addons');
```

to

``` js
var React = require('react');
```

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,6 @@
 node_modules
 bower_components
 example/*
+.idea
```
Comment (rsahlstrom) [inline .gitignore]: This looks to be something specific to your environment and should probably go in your global .gitignore file and not in the project specific one.

https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

**Comment by rsahlstrom** [comment]
/example/index.js was probably not commit due to the .gitignore file. We probably don't need example/\* in the .gitignore anyways.

File: .gitignore (hunk 2 above)
Comment (dahei) [inline .gitignore]: Yes, you are right.

File: src/mixins/ClassNameMixin.jsx (hunk 1 above)
Comment (dahei) [inline src/mixins/ClassNameMixin.jsx:46]: Agreed.

**Comment by dahei** [comment]
Yes, the example/index.js wasn't committed because of the .gitignore. I think we should remove it from there.

**Comment by dahei** [comment]
I fixed the require and gitignore issues and committed a new build. Please have a look.

**Comment by rsahlstrom** [comment]
Oh man, sorry to leave you hanging for three years. Archiving project.

**Pull request closed without merging**